hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkUtils.scala

```diff
@@ -95,17 +96,38 @@ object HoodieSparkUtils {
     createRdd(df, avroSchema, structName, recordNamespace)
   }
 
-  def createRdd(df: DataFrame, avroSchema: Schema, structName: String, recordNamespace: String)
+  def createRdd(df: DataFrame, latestSchema: Schema, structName: String, recordNamespace: String): RDD[GenericRecord] = {
+    val avroSchema = AvroConversionUtils.convertStructTypeToAvroSchema(df.schema, structName, recordNamespace)
+    // if schema generated from df.schema is same as latest schema, no special handling is required.
+    if(TableSchemaResolver.isSchemaEquals(avroSchema, latestSchema)) {
+      createRdd(df, avroSchema, null, structName, recordNamespace)
+    } else { // if not, it means that table schema got evolved, but this batch of records were generated with an older
+      // schema.
+      createRdd(df, avroSchema, latestSchema, structName, recordNamespace)
+    }
+  }
+
+  def createRdd(df: DataFrame, avroSchema: Schema, latestSchema: Schema, structName: String, recordNamespace: String)
```

Comment:
Is the scope of this method to be private ? If yes, can you change it so this method is not invoked in other code paths